### PR TITLE
MAKE-687: use common queue caching mechanism

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: queue/group_utils.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+type cacheItem struct {
+	q    amboy.Queue
+	ttl  time.Duration
+	ts   time.Time
+	name string
+}
+
+func (c *cacheImpl) Len() int {
+	c.mu.Lock()
```
Comment (bsamek) [inline queue/group_utils.go]: I think this can be an RLock()

File: queue/group_utils.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	defer c.mu.Unlock()
+
+	return len(c.q)
+}
+
+func (c *cacheImpl) Names() []string {
+	c.mu.Lock()
```
Comment (bsamek) [inline queue/group_utils.go]: RLock()

File: queue/group_utils.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+					if item.ttl > 0 && item.ttl > time.Since(item.ts) {
+						continue
+					}
+
+					if item.q.Stats().IsComplete() {
+						item.q.Runner().Close(ctx)
+
+						wait := make(chan struct{})
+						func() {
```
Comment (bsamek) [inline queue/group_utils.go]: Did you mean to put this in a goroutine?

Comment (tychoish) [reply queue/group_utils.go]: nope, want to be able to lock and use a defer but have it happen synchronously.

Comment (tychoish) [reply queue/group_utils.go]: jk, you're right.

**Pull request merged**